do-release-candidate.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/sh -ex
```

Comment:
I figure I'll make a copy for now, as `do-release.sh` is already complicated and I'd rather not make changes to it yet. We should be able to merge after we get rid of `latest_release` non-sense (see #903).